src/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -12,20 +12,22 @@ export interface VideoPlayerProps {
   className?: string;
   /** Optional string to set the text for a video description */
   descriptionText?: string;
+  /** Optional string to set a code snippet provided by YouTube or Vimeo; the `videoPlayer` component will accept the `embedCode` prop or the `videoId` and `videoType` props */
+  embedCode?: string;
```

Comment:
Should this be a string or more restrictive to only allow an HTML iframe element?

Reply:
@nypl-wluisi There are a number of checks built into the component aiming to validate that an iframe is included in the string.

@EdwinGuzman Can this prop be typed in a way to validate an HTML string is being passed?

Reply:
Yes,

```
embedCode?: `<iframe${string}`;
```

That might be enough to start off the string with `"<iframe"` but not enough if you want to keep track of the element attributes. That might be trickier because the order might be different every time and that's as much of a regex you can do in a type string.

Here, I mistyped "iframe" and vscode found the type error:

<img width="828" alt="Screen Shot 2021-12-22 at 12 37 11 PM" src="https://user-images.githubusercontent.com/1280564/147133170-a3e64a5c-8f91-428d-b320-e54f102fe388.png">

Reply:
This change is causing an error when pushing the code to github, so I will leave it out for now.  There are other checks built into the component that will do similar validation.